kornia/geometry/bbox_v2.py

```diff
@@ -0,0 +1,585 @@
+            raise ValueError(f"Boxes shape must be (N, 4, 2) or (B, N, 4, 2). Got {quadrilaterals.shape}.")
+
+        self._boxes = quadrilaterals
+
+    def boxes_shape(self) -> Tuple[torch.Tensor, torch.Tensor]:
+        r"""Compute bounding boxes heights and widths.
+
+        Returns:
+            - Bounding box heights, shape of :math:`(N,)` or :math:`(B,N)`.
+            - Boundingbox widths, shape of :math:`(N,)` or :math:`(B,N)`.
+
+        Example:
+            >>> boxes_xyxy = torch.tensor([[[
+            ...     [1., 1.],
+            ...     [2., 1.],
+            ...     [2., 2.],
+            ...     [1., 2.],
+            ... ], [
+            ...     [1., 1.],
+            ...     [3., 1.],
+            ...     [3., 2.],
+            ...     [1., 2.],
+            ... ]]])  # 1x2x4x2
+            >>> boxes = Boxes.from_tensor(boxes_xyxy)
+            >>> boxes.boxes_shape()
+            (tensor([[1., 1.]]), tensor([[1., 2.]]))
+        """
+        boxes_xywh = self.to_tensor(mode='xywh')
+        widths, heights = boxes_xywh[..., 2], boxes_xywh[..., 3]
+        return heights, widths
+
+    @classmethod
+    def from_tensor(cls, boxes: torch.Tensor, mode: str = "xyxy") -> "Boxes":
+        r"""Convert 2D bounding boxes to kornia format according to the format in which the boxes are provided.
+
+        Args:
+            boxes: 2D boxes to be transformed, shape of :math:`(N,4)` or :math:`(B,N,4)`.
+            mode: The format in which the boxes are provided.
+
+                * 'xyxy': boxes are assumed to be in the format ``xmin, ymin, xmax, ymax`` where ``width = xmax - xmin``
+                  and ``height = ymax - ymin``.
+                * 'xyxy_plus_1': like 'xyxy' where ``width = xmax - xmin + 1`` and  ``height = ymax - ymin + 1``.
+                * 'xywh': boxes are assumed to be in the format ``xmin, ymin, width, height`` where
+                  ``width = xmax - xmin`` and ``height = ymax - ymin``.
+                * 'xywh_plus_1': like 'xywh' where ``width = xmax - xmin + 1`` and  ``height = ymax - ymin + 1``.
+
+        Returns:
+            2D Bounding boxes tensor in kornia format, shape of :math:`(N, 4, 2)` or :math:`(B, N, 4, 2)` and dtype of
+            ``boxes`` if it's a floating point data type and ``float`` if not.
+
+        Examples:
+            >>> boxes_xyxy = torch.as_tensor([[0, 3, 1, 4], [5, 1, 8, 4]])
+            >>> Boxes.from_tensor(boxes_xyxy, mode='xyxy')
+        """
+        if not (2 <= boxes.ndim <= 3 and boxes.shape[-1] == 4):
+            raise ValueError(f"Boxes shape must be (N, 4) or (B, N, 4). Got {boxes.shape}.")
+
+        batched = boxes.ndim == 3
+        boxes = boxes if batched else boxes.unsqueeze(0)
+        boxes = boxes if boxes.is_floating_point() else boxes.float()
+
+        xmin, ymin = boxes[..., 0], boxes[..., 1]
+        mode = mode.lower()
+        if mode in ("xyxy", "xyxy_plus_1"):
```

Comment:
Okay for now. Probably add a few enums later.

Reply:
I can do it as torchscript has support for enums. However, I'm not sure if it would support passing an string with the enum field name like it's done in transforms. It would be okey if we force the user to always use the enum class?

Reply:
I tried to add enums. I face a limitation. Torchscript isn't aware of classmethod attached to enums. So, I can do `BoxMode.get(mode)` like in other parts of kornia. So, if we use enums here, users won't be able to pass a string. Do we want this behavior?